libs/remix-ui/workspace/src/lib/actions/index.ts

```diff
@@ -44,19 +59,53 @@ export const initWorkspace = (filePanelPlugin) => async (reducerDispatch: React.
       dispatch(setCurrentWorkspace('code-sample'))
       const filePath = await loadWorkspacePreset('code-template')
       plugin.on('editor', 'editorMounted', async () => await plugin.fileManager.openFile(filePath))
-    } else {
-      if (workspaces.length === 0) {
-        await createWorkspaceTemplate('default_workspace', 'remixDefault')
-        plugin.setWorkspace({ name: 'default_workspace', isLocalhost: false })
-        dispatch(setCurrentWorkspace('default_workspace'))
-        await loadWorkspacePreset('remixDefault')
-      } else {
-        if (workspaces.length > 0) {
-          workspaceProvider.setWorkspace(workspaces[workspaces.length - 1])
-          plugin.setWorkspace({ name: workspaces[workspaces.length - 1], isLocalhost: false })
-          dispatch(setCurrentWorkspace(workspaces[workspaces.length - 1]))
+    } else if (window.location.pathname && window.location.pathname !== '/') {
+      const route = window.location.pathname
+      if (route.startsWith('/address/0x') && route.length === 51) {
+        const contractAddress = route.split('/')[2]
+        let data
+        try {
+          const etherscanKey = await plugin.call('config', 'getAppParameter', 'etherscan-access-token')
```

Comment:
for this kind of feature it'd be better to provide a key. I can provide you one.

Reply:
I was thinking same unless API key is exploited